packages/template-retail-react-app/app/commerce-api/hooks/useShopper.js

```diff
@@ -58,9 +58,16 @@ const useShopper = (opts = {}) => {
         }
     }, [customer.authType, basket.loaded])
 
-    // Call merge basket whenever user type changes from guest to registered
+    // Call merge basket.
     useEffect(() => {
-        if (customer.authType === 'registered') {
+        // Only call merge when there are items in the guest basket and you are
+        // a returning customer.
+        const shouldMerge =
+            customer.authType === 'registered' &&
+            !customer.isNewlyRegistered &&
+            basket.itemCount > 0
+
+        if (shouldMerge) {
             basket.mergeBasket()
         }
     }, [customer.authType])
```

Comment:
```suggestion
    }, [customer.authType, customer.isNewlyRegistered, basket.itemCount])
```
Are these needed? (React hooks are still confusing!)

Reply:
No. If we added those values it just means that the hooks `didUpdate` logic would be run when those addition values change, which isn't necessarily what we want to be doing. I know the hook with the guard in it is ugly, there are alternatives but they are equally unsightly.